projects/plugins/jetpack/extensions/blocks/map/view/mapkit.js

```diff
@@ -99,13 +99,32 @@ class MapkitBlock {
 				point.coordinates.latitude,
 				point.coordinates.longitude
 			);
-			const annotation = new this.mapkit.MarkerAnnotation( coordinate, {
+			return new this.mapkit.MarkerAnnotation( coordinate, {
 				color: this.color,
+				title: point.title,
+				callout: {
+					calloutContentForAnnotation: function ( annotation ) {
+						const element = document.createElement( 'div' );
+						const title = element.appendChild( document.createElement( 'div' ) );
+						title.style.fontSize = '17px';
+						title.style.fontWeight = '500';
+						title.style.lineHeight = '17px';
+						title.style.marginTop = '4px';
+						title.textContent = annotation.title;
+
+						const caption = element.appendChild( document.createElement( 'p' ) );
+						caption.style.fontSize = '14px';
+						caption.style.margin = '0px 0px 4px 0px';
+						caption.textContent = annotation.data?.caption;
```

Comment:
To combat long captions, setting `maxWidth` to `50%` doesn't work, but setting `width` to `(annotation.element.parentElement.offsetWidth / 2) + "px"` does.

**Before:**
<img width="747" alt="image" src="https://github.com/Automattic/jetpack/assets/1287077/b680efc9-9303-43d9-9d39-a6a008df1d65">

**After:**
<img width="745" alt="image" src="https://github.com/Automattic/jetpack/assets/1287077/dd43b140-f209-43bf-875d-6ec84404126c">

Reply:
Thanks!